src/Aspire.Dashboard/Utils/FormatHelpers.cs

```diff
@@ -1,55 +1,77 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Collections.Concurrent;
 using System.Globalization;
 using System.Text.RegularExpressions;
 
 namespace Aspire.Dashboard.Utils;
 
 internal static partial class FormatHelpers
 {
-    static string GetLongTimePatternWithMilliseconds()
+    private sealed record MillisecondFormatStrings(string LongTimePattern, string ShortDateLongTimePattern);
+
+    // Use culture name as key. Culture name isn't case sensitive.
+    private static readonly ConcurrentDictionary<string, MillisecondFormatStrings> s_formatStrings = new(StringComparer.OrdinalIgnoreCase);
+
+    [GeneratedRegex("(:ss|:s)")]
+    private static partial Regex MatchSecondsInTimeFormatPattern();
+
+    private static MillisecondFormatStrings GetMillisecondFormatStrings(CultureInfo cultureInfo)
     {
-        // From https://learn.microsoft.com/dotnet/standard/base-types/how-to-display-milliseconds-in-date-and-time-values
+        // The culture name is used as the key for the cache. The name is used to avoid unbounded growth from custom CultureInfo instances.
+        // Note that the culture name must match to a cached readonly instance of CultureInfo.
+        // If the passed in culture info has customizations then they won't be used. The cached CultureInfo with a name is the source of truth.
```

Comment:
Does this actually respect the user's preferences then? If I'm in en-AU but select a different format in the OS, my preference won't be honoured and I'll be forced back to my culture's default. That seems wrong to me.

In hosted scenarios there may not be a way around this, but it would be a shame to force this on users in local dev.

The original bug from @timheuer related to customized time formats. I wonder if this would revert that fix.

Reply:
I just tested changing my OS to show 24 hour instead of 12 hour and it worked. At least locally.

In internet scenarios only the culture code is sent to the server (e.g. en-US, en-NZ, fr-FR, etc) via https://github.com/dotnet/aspire/pull/2192

Reply:
Thanks, it's the local scenario I was concerned about. I suppose this means that if I, in `en-AU`, modify my OS-level preferences, then the `en-AU` locale on my machine is modified accordingly?

Reply:
Yes